storage/src/main/java/tech/pegasys/teku/storage/store/Store.java

```diff
@@ -72,8 +72,12 @@ class Store implements UpdatableStore {
 
   private final int hotStatePersistenceFrequencyInEpochs;
 
+  private final ReadWriteLock votesLock = new ReentrantReadWriteLock();
+  private final Lock readVotesLock = votesLock.readLock();
```

Comment:
Maybe for a follow up PR but I wonder if we should be splitting vote tracking into a separate class rather than having it in `Store`. I think (not sure) we already use a specific VoteUpdater interface for votes anyway so the concerns are pretty separate, would be interesting to see if we can make them completely separate.